source/server/server.cc

```diff
@@ -128,19 +128,62 @@ void InstanceImpl::drainListeners() {
   drain_manager_->startDrainSequence(nullptr);
 }
 
-void InstanceImpl::failHealthcheck(bool fail) {
-  // We keep liveness state in shared memory so the parent process sees the same state.
-  server_stats_->live_.set(!fail);
-}
+void InstanceImpl::failHealthcheck(bool fail) { server_stats_->live_.set(!fail); }
+
+// Local implementation of Stats::MetricSnapshot used to flush metrics to sinks. We could
+// potentially make this static and have a clear() method to avoid some vector constructions
```

Comment:
What you are talking about here is making the array member variables in this class static -- or the class instance static -- is that right?

In that case I agree -- let's wait for an indication this is meaningful in perf. It would be good to have some perf metrics for stats sinks though.

For the comment above, can you do:
  s/make this static/have a single class instance held in a static/
or similar? At first I thought you meant  having the class methods be static.


However, I'd vote for the pattern where we iterate using a lambda over the pattern of collecting the vectors during construction. But you could leave a TODO if you agree and follow-up? I think that would work nicely in  
UdpStatsdSink::flush, anyway.

And it would be easy to supply a utility function -- either in the test/ hierarchy or in source/common/stats/utility.h -- to capture the results of the lambda in a vector and return it, at least for easy use in tests?

Seems like that saves some memory churn, is pretty fast, and not too hard to use at least for the use-case here. 
Actually come to think of it I'd probably carry that through to Store::counters() as well....but not in this PR.

Reply:
Yeah let me change the comment and add a TODO. The only downside I see of the lambda approach that I will still have to create a vector of the snapped latched values so that I can pass them to the lambda callback, but that's still better than what we have here.